### ISSUE TITLE
Fresh installation does not work on cleanly installed Ubuntu1804

### ISSUE THREAD

**Issue by gggh000** [onnx/onnx#3621]
# Bug Report

Plain installation onto Ubuntu 1804  failed. Can not believe it is failing at  such basic level:
Instruction: 
Official Python packages
ONNX released packages are published in PyPi.

pip install numpy protobuf==3.16.0
pip install onnx

Following the instruction: 

```
root@nonroot-MS-7B22:~/superbenchmark# pip3 install numpy protobuf==3.16.0
Requirement already satisfied: numpy in /usr/local/lib/python3.6/dist-packages
Collecting protobuf==3.16.0
  Downloading https://files.pythonhosted.org/packages/e0/c3/b1a51b89c36122c58635143058fcf4a980b0c97db7ee856389330d430c02/protobuf-3.16.0-cp36-cp36m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.0MB)
    100% |████████████████████████████████| 1.0MB 1.3MB/s 
Requirement already satisfied: six>=1.9 in /usr/lib/python3/dist-packages (from protobuf==3.16.0)
Installing collected packages: protobuf
  Found existing installation: protobuf 3.0.0
    Not uninstalling protobuf at /usr/lib/python3/dist-packages, outside environment /usr
Successfully installed protobuf-3.16.0
root@nonroot-MS-7B22:~/superbenchmark# pip3 install nnx
Collecting nnx

root@nonroot-MS-7B22:~/superbenchmark# pip3 install onnx
Collecting onnx
  Downloading https://files.pythonhosted.org/packages/cf/98/82671f253342463e63fa972713f2bab1d936dd178455501d36ff288b209f/onnx-1.10.0.tar.gz (10.0MB)
    100% |████████████████████████████████| 10.0MB 160kB/s 
    Complete output from command python setup.py egg_info:
    fatal: not a git repository (or any of the parent directories): .git
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-zma2mwjq/onnx/setup.py", line 318, in <module>
        raise FileNotFoundError("Unable to find " + requirements_file)
    FileNotFoundError: Unable to find requirements.txt


```

**Comment by gggh000** [comment]
not sure if i understood. are you telling maintainer to include the req. txt?

**Comment by jcwchen** [comment]
Hi @gggh000,
Thank you for reporting this. Yes requirements.txt was introduced in 1.10.0 and that file was not included in the 1.10.0 source distribution. If there is no prebuilt wheel from PyPI for your environment, building from source distribution will bump into this issue. 

If you are urgent to use 1.10, you can use this updated [source distribution](https://github.com/onnx/onnx/files/6911243/onnx-1.10.0.tar.gz) by `pip install onnx-1.10.0.tar.gz`. We will soon release a patch to fix this. Sorry for the inconvenience and thank you again for catching this issue!

**Comment by gggh000** [comment]
I will try this your link, thx

**Comment by rajeevsrao** [comment]
@gggh000 thanks, this is also fixed in v1.10.1: https://github.com/onnx/onnx/releases/tag/1.10.1

**Comment by gggh000** [comment]
I dont know how to install this one. What do you after downloading tar bz?
Becuse I downloaded source and now build fails with following even though installed protobuf.

Do you check or run sanity test before releasing? Becayuse your stuff appears seriously broken and does not look like in a state where public should access such a broken stuff.

```

CMake Error at CMakeLists.txt:295 (message):
  Protobuf compiler not found
Call Stack (most recent call first):
  CMakeLists.txt:326 (relative_protobuf_generate_cpp)


-- Configuring incomplete, errors occurred!
See also "/root/onnx/build_source/CMakeFiles/CMakeOutput.log".
(venv) root@nonroot-MS-7B22:~/onnx/build_source# pip install numpy protobuf==3.16.0
Requirement already satisfied: numpy in /usr/local/lib/python3.6/dist-packages
Requirement already satisfied: protobuf==3.16.0 in /usr/local/lib/python3.6/dist-packages
Requirement already satisfied: six>=1.9 in /usr/lib/python3/dist-packages (from protobuf==3.16.0)
(venv) root@nonroot-MS-7B22:~/onnx/build_source# pip install numpy protobuf
Requirement already satisfied: numpy in /usr/local/lib/python3.6/dist-packages
Requirement already satisfied: protobuf in /usr/local/lib/python3.6/dist-packages
Requirement already satisfied: six>=1.9 in /usr/lib/python3/dist-packages (from protobuf)
(venv) root@nonroot-MS-7B22:~/onnx/build_source# pip install numpy protobuf==3.16.0^C

```

**Comment by gggh000** [comment]
I am wasting all my time for fixing all your garbage stuff that it should be taken care of before even releasing.

**Comment by rajeevsrao** [comment]
A reminder that we welcome all feedback and contributions, but in keeping with https://onnx.ai/codeofconduct.html

ONNX CI/CD pipeline is here. https://github.com/onnx/onnx/actions

**Comment by gggh000** [comment]
What are you talkin about? I am pointing of your failures which i dont have to and now you are reminding me a code of conduct??? Conduct appears to be more appeopriate for you.

**Comment by jcwchen** [comment]
I spent time building a fresh Ubuntu 18.04 and followed [README.md](https://github.com/onnx/onnx/blob/master/README.md) for building ONNX, but no matter how I could not repro your cmake error @gggh000. Your log is incomplete so I cannot tell whether you followed the right steps. As you can see, the error showed that it cannot find protobuf compiler so you need to build protobuf first before building ONNX.

Protbuf installation on Linux is mentioned in the README.md:
```
git clone https://github.com/protocolbuffers/protobuf.git
cd protobuf
git checkout v3.16.0
git submodule update --init --recursive
mkdir build_source && cd build_source
cmake ../cmake -Dprotobuf_BUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_POSITION_INDEPENDENT_CODE=ON -Dprotobuf_BUILD_TESTS=OFF -DCMAKE_BUILD_TYPE=Release
make -j$(nproc)
make install
```
I don't know what the error this line generated: `cmake ../cmake -Dprotobuf_BUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_POSITION_INDEPENDENT_CODE=ON -Dprotobuf_BUILD_TESTS=OFF -DCMAKE_BUILD_TYPE=Release`. Could you please point out? We are eager to enhance the build document to improve developers' experience so any valid feedback is appreciated.

After installing protobuf correctly (you can use `protoc --version` to verify), you should be able to build ONNX from source, or `pip install onnx-1.10.0.tar.gz` which I gave you, or just `pip install onnx` because we just released a patch 1.10.1 yesterday to fix the source distribution issue. Besides, If you don't understand how to build protobuf, there is another option to install/use ONNX by conda-forge.

We are all willing to help if you raise any issue. Every time you bump into an issue, please don't hesitate to let us know first to save your time. Thanks

**Comment by askhade** [comment]
The project is governed and created by the community which is open for contributions for improvement. Please adhere to the code of conduct https://onnx.ai/codeofconduct.html during every communication.
